core/src/language/python.rs

```diff
@@ -0,0 +1,836 @@
+use crate::parser::ParsedData;
+use crate::rust_types::{
+    RustEnumShared, RustEnumVariantShared, RustItem, RustType, RustTypeFormatError, SpecialRustType,
+};
+use crate::topsort::topsort;
+use crate::{
+    language::Language,
+    rust_types::{RustEnum, RustEnumVariant, RustField, RustStruct, RustTypeAlias},
+};
+use once_cell::sync::Lazy;
+use std::collections::{BTreeSet, HashSet};
+use std::hash::Hash;
+use std::{collections::HashMap, io::Write};
+
+use super::CrateTypes;
+
+use convert_case::{Case, Casing};
+// Collect unique type vars from an enum field
+// Since we explode enums into unions of types, we need to extract all of the generics
+// used by each individual field
+// We do this by exploring each field's type and comparing against the generics used by the enum
+// itself
+fn collect_generics_for_variant(variant_type: &RustType, generics: &[String]) -> Vec<String> {
```

Comment:
Nit: consider using a wrapper that pushes to an `&mut Vec<String>` here, to avoid the deep recursive frequent allocations and data copying between vectors. Also consider returning a `HashSet` or `BTreeSet` from this function.